solidity-v1/dashboard/src/hooks/useAutoConnect.js

```diff
@@ -41,37 +44,46 @@ const useAutoConnect = () => {
 
   useEffect(() => {
     if (injectedTried) return
-    injected.getAccounts().then((accounts) => {
-      setInjectedTried(true)
-      if (
-        (!isEmptyArray(accounts) && isExactRoutePath) ||
-        isWalletFromUrlSameAsInMetamask(accounts)
-      ) {
-        connectAppWithWallet(injected, false).catch((error) => {
-          // Just log an error, we don't want to do anything else.
-          console.log(
-            "Eager injected connector cannot connect with the dapp:",
-            error.message
-          )
-        })
-      } else if (walletAddressFromUrl && !connector) {
-        const explorerModeConnector = new ExplorerModeConnector()
-        openModal(MODAL_TYPES.ExplorerMode, {
-          connectAppWithWallet,
-          connector: explorerModeConnector,
-          address: walletAddressFromUrl,
-          connectWithWalletOnMount: true,
-        })
-      }
-    })
+
+    const connectWallet = (connector) => {
+      connector.getAccounts().then((accounts) => {
+        setInjectedTried(true)
+        if (
+          (!isEmptyArray(accounts) && isExactRoutePath) ||
+          isWalletFromUrlSameAsInProvider(accounts)
+        ) {
+          connectAppWithWallet(connector, false).catch((error) => {
+            // Just log an error, we don't want to do anything else.
+            console.log(
+              "Eager injected connector cannot connect with the dapp:",
+              error.message
+            )
+          })
+        } else if (walletAddressFromUrl && !connector) {
+          const explorerModeConnector = new ExplorerModeConnector()
+          openModal(MODAL_TYPES.ExplorerMode, {
+            connectAppWithWallet,
+            connector: explorerModeConnector,
+            address: walletAddressFromUrl,
+            connectWithWalletOnMount: true,
+          })
+        }
+      })
+    }
```

Comment:
Right now Explorer Mode does not work as it supposed to work - when we change the wallet address in the url for the one that is not connected to neither MetaMask nor Tally, then nothing happens (and it should connect to Explorer Mode in such case)

The reason of such behavious lies in this function. The `connector` variable that is the argument of `connectWallet` function is not the same  as `connector` used in `else if` statement. The first one is the actual connector that we created for MetaMask/Tally, and the seconds one is the actual used connector that is saved on Redux side. You can see in the deleted lines above that we used `injected` and `connector` as two different variables.

So we have to change those names to different ones. I think we can just change the first `connector` variable to `injectedConnector` but I'm open to suggestion if you have better idea for the name or want to rename the other one.

```suggestion
    const connectWallet = (injectedConnector) => {
      injectedConnector.getAccounts().then((accounts) => {
        setInjectedTried(true)
        if (
          (!isEmptyArray(accounts) && isExactRoutePath) ||
          isWalletFromUrlSameAsInProvider(accounts)
        ) {
          connectAppWithWallet(injectedConnector, false).catch((error) => {
            // Just log an error, we don't want to do anything else.
            console.log(
              "Eager injected connector cannot connect with the dapp:",
              error.message
            )
          })
        } else if (walletAddressFromUrl && !connector) {
          const explorerModeConnector = new ExplorerModeConnector()
          openModal(MODAL_TYPES.ExplorerMode, {
            connectAppWithWallet,
            connector: explorerModeConnector,
            address: walletAddressFromUrl,
            connectWithWalletOnMount: true,
          })
        }
      })
    }
```

Reply:
This is awesome! This fix is 🔥

I just typed it in and it worked! 💪